tools/protodoc/protodoc.bzl

```diff
@@ -0,0 +1,56 @@
+# Borrowed from
+# https://github.com/bazelbuild/rules_go/blob/master/proto/toolchain.bzl. This
+# does some magic munging to remove workspace prefixes from output paths to
+# convert path as understood by Bazel into paths as understood by protoc.
+def _proto_path(proto):
+    """
+    The proto path is not really a file path
+    It's the path to the proto that was seen when the descriptor file was generated.
+    """
+    path = proto.path
+    root = proto.root.path
+    ws = proto.owner.workspace_root
+    if path.startswith(root): path = path[len(root):]
+    if path.startswith("/"): path = path[1:]
+    if path.startswith(ws): path = path[len(ws):]
+    if path.startswith("/"): path = path[1:]
+    return path
+
+def _proto_doc_aspect_impl(target, ctx):
+    transitive_outputs = depset()
```

Comment:
can you add some high level comment to this function to explain the flow/logic? I have no idea about this skylark thingy, but if I have to tweak/hack it, it would be nice to know whats going on here.

Reply:
Docs added, LMK if there is anything extra you want. Happy to have you contribute, I'm going to get a simple example working for one of the leaf `.proto` (e.g. `base.proto`) and then we can divvy up the work once we're happy with that PR.